manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -34,6 +42,12 @@ spec:
       serviceAccountName: {{ include "gateway.serviceAccountName" . }}
       {{- if not .Values.runAsRoot }}
       securityContext:
+        {{- if $unprivPort }}
```

Comment:
I would set this if "semverCompare > 1.22" - regardless of the values.unpriviledged port.
Simpler, no harm in having this.

Reply:
It is also consistent with the auto-created ones.